### ISSUE TITLE
Marktext should not renumber existing numbered lists

### ISSUE THREAD

**Issue by N0rbert** [marktext/marktext#2035]
### Description

Using MarkText with version control system become difficult because of renumbering lists.
### Steps to reproduce

1. Have simple Markdown document with numbered list inside:

       Some text before list:
       
       1. First item
       1. Second item
       1. Third item

       Some text after list.

2. Open this document using MarkText
3. Save the document

**Expected behavior:**

the MarkText does not change the document

**Actual behavior:**

the MarkText changed the document contents  and replaced second "1." and third "1." to "2." and "3.":

```
Some text before list:

1. First item
2. Second item
3. Third item

Some text after list.
```

this behavior is unexpected and messes diifs in the version control systems. And it causes `mdl` to produce [`MD029 Ordered list item prefix`](https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md#md029---ordered-list-item-prefix) rule violation.

### Versions

- Mark Text: v0.16.1 AppImage
- Operating system: Ubuntu MATE 16.04 LTS

**Comment by fxha** [comment]
I'm closing this issue in favour of #2189. You may want to subscribe there for further updates.

**Comment by N0rbert** [comment]
Still not fixed in 0.16.3.

**Comment by N0rbert** [comment]
Still not fixed in v0.17.0-rc.1.
